kornia/geometry/transform/homography_warper.py

```diff
@@ -164,6 +168,109 @@ def homography_warp3d(
     return F.grid_sample(patch_src, warped_grid, mode=mode, padding_mode=padding_mode, align_corners=align_corners)
 
 
+class Homography(nn.Module):
+    def __init__(self):
+        """Module to be used together with ImageRegistrator module for the optimization-based image
+        registration."""
+        super().__init__()
+        self.model = nn.Parameter(torch.Tensor(3, 3))
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        torch.nn.init.eye_(self.model)
+
+    def forward(self):
+        return torch.unsqueeze(self.model, dim=0)  # 1x3x3
+
+    def get_inverse_model(self):
+        return torch.unsqueeze(torch.inverse(self.model), dim=0)
+
+
+class ImageRegistrator(nn.Module):
+    """'Module, which performs optimization-based image registration, similar to https://github.com/kornia/kornia-
+    examples/blob/master/homography.ipynb."""
+    def __init__(self,
+                 model_type: str = 'H',
+                 optimizer: torch.optim.Optimizer = optim.Adam,
+                 loss_fn = F.l1_loss,
+                 pyr_levels = 5,
+                 lr: float = 1e-3,
+                 n_iter: int = 100):
+        super().__init__()
+        self.model_type = model_type
+        if model_type == "H":
```

Comment:
what other models are you thinking about ?

Reply:
Thin plate and other elastics, for example. 
@Borda, could you please give us a hint of a models, which are used in medical image registration?

Reply:
yes, the common request is a Thin plate as it is not so demanding on computation
the second common is B-spline

Reply:
Thank you!